apps/highlight/src/highlight/signatures.ts

```diff
@@ -102,22 +145,37 @@ async function verifyEip1271SignatureWithAbi(
 }
 
 export async function verifySignature(
-  domain: Required<TypedDataDomain>,
+  domain: TypedDataDomain | StarknetDomain,
```

Comment:
Why did we drop `Required` there?  Before we didn't need to do all those `as Required<TypedDataDomain>` below, but now we do because our input is more relaxed than our functions require. Maybe they should all stay `Required` if we cast those anyway?